src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -187,18 +188,33 @@ private void DecrementTermReferenceCount(LowLevelTransaction llt, long keyContai
                 throw new VoronErrorException("A term exists without any references? That should be impossible");
             if (termRefCount == 1) // no more references, can delete
             {
+                term[0] = 15 & 0xF;
                 Container.Delete(llt, state.TermsContainerId, keyContainerId);
                 return;
             }
 
             term[0] = (byte)((term[0] & 0xF0) | (termRefCount - 1));
         }
 
-
-        private void IncrementTermReferenceCount(LowLevelTransaction llt, long keyContainerId)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Span<byte> GetTermReference(LowLevelTransaction llt, long keyContainerId, out int termRefCount)
         {
             var term = Container.GetMutable(llt, keyContainerId);
-            int termRefCount = term[0] & 0xF;
+            termRefCount = term[0] & 0xF;
+            return term;
+        }
+        
+        private int IncrementTermReferenceCount(LowLevelTransaction llt, long keyContainerId, bool conditionalIncrease = false)
+        {
+            var term = GetTermReference(llt, keyContainerId, out var termRefCount);
+            
+            
+            //In case of calling prevent deletion that means we're handling split/merges which copy the data.
+            //Let's increase it only when termRefCount is equal to 1 to avoid accidentally removal.
+            if (conditionalIncrease && termRefCount > 1)
```

Comment:
Let's talk directly about that. I can't really understand the purpose of that. Maybe it's just a matter of better naming of `conditionalIncrease` variable and better description

Reply:
I don't follow this either.